cmd/cli/app/auth/offline_use.go

```diff
@@ -0,0 +1,112 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/stacklok/minder/internal/config"
+	clientconfig "github.com/stacklok/minder/internal/config/client"
+	"github.com/stacklok/minder/internal/util"
+)
+
+// offlineTokenUseCmd represents the offline-token use command
+var offlineTokenUseCmd = &cobra.Command{
+	Use:   "use",
+	Short: "Use an offline token",
+	Long: `The minder auth offline-token use command project lets you install and use an offline token
+for the minder control plane.
+
+Offline tokens are used to authenticate to the minder control plane without
+requiring the user's presence. This is useful for long-running processes
+that need to authenticate to the control plane.`,
+
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		if err := viper.BindPFlags(cmd.Flags()); err != nil {
+			return fmt.Errorf("error binding flags: %s", err)
+		}
+
+		clientConfig, err := config.ReadConfigFromViper[clientconfig.Config](viper.GetViper())
+		if err != nil {
+			return fmt.Errorf("error reading config: %w", err)
+		}
+
+		f := viper.GetString("offline-token-path")
+		tok := viper.GetString("offline-token")
+		if tok == "" {
+			fpath := filepath.Clean(f)
+			tokbytes, err := os.ReadFile(fpath)
+			if err != nil {
+				return fmt.Errorf("error reading file: %w", err)
+			}
+
+			tok = string(tokbytes)
+		}
+
+		// No longer print usage on returned error, since we've parsed our inputs
+		// See https://github.com/spf13/cobra/issues/340#issuecomment-374617413
+		cmd.SilenceUsage = true
+
+		issuerUrlStr := clientConfig.Identity.CLI.IssuerUrl
+		clientID := clientConfig.Identity.CLI.ClientId
+
+		creds, err := util.RefreshCredentials(tok, issuerUrlStr, clientID)
+		if err != nil {
+			return fmt.Errorf("couldn't fetch credentials: %v", err)
+		}
+
+		// save credentials
+		filePath, err := util.SaveCredentials(util.OpenIdCredentials{
+			AccessToken:          creds.AccessToken,
+			RefreshToken:         creds.RefreshToken,
+			AccessTokenExpiresAt: creds.AccessTokenExpiresAt,
+		})
+		if err != nil {
+			cmd.PrintErrf("couldn't save credentials: %s\n", err)
+		}
+
+		cmd.Printf("Your access credentials have been saved to %s\n", filePath)
+
+		return nil
+	},
+}
+
+func init() {
+	OfflineTokenCmd.AddCommand(offlineTokenUseCmd)
+
+	offlineTokenUseCmd.Flags().StringP("offline-token-path", "f", "offline.token", "The file that contains the offline token")
+	offlineTokenUseCmd.Flags().StringP("offline-token", "t", "",
```

Comment:
Should we rename these to `token` and `token-path` instead? The command is already called "offline-token" and they are available only in it, so we can avoid repeating `offline` in the flag names.